### PR TITLE
fix(RHTAPBUGS-498): set application as owner of SEB

### PR DESCRIPTION
**Pull request by dirgim**
* If environment is not ephemeral, set application as owner
* The SEBs for ephemeral Envs are expected to be cleaned up along with the Environment
* All other SEBs are meant to persist and be deleted with the Application

### CODE REVIEW

**Comment by MartinBasti** [comment]
The commit (and code) misses explanation why

File: controllers/snapshot/snapshot_adapter.go [hunk 1]
```diff
@@ -501,10 +501,18 @@ func (a *Adapter) createSnapshotEnvironmentBindingForSnapshot(application *appli
 		environment.Name,
 		snapshot, *components)
 
-	// set environment as owner of snapshotEnvironmentBinding on controlled
-	err := ctrl.SetControllerReference(environment, snapshotEnvironmentBinding, a.client.Scheme())
-	if err != nil {
-		return nil, err
+	// If ephemeral, set environment as owner of snapshotEnvironmentBinding on controlled,
```
Comment (MartinBasti) [inline controllers/snapshot/snapshot_adapter.go:504]: This comment should explain why instead of rephrasing what's written in the code

Comment (dirgim) [reply controllers/snapshot/snapshot_adapter.go]: Ack, updated.

**Review by MartinBasti — APPROVED** [review]

**Review by jsztuka — APPROVED** [review]
lgtm

**Review by dheerajodha — APPROVED** [review]
lgtm

**Review by hongweiliu17 — COMMENTED** [review]
LGTM

**Comment by hongweiliu17** [comment]
Thanks for fixing it!

**Comment by mshaposhnik** [comment]
This PR conflicts with https://issues.redhat.com/browse/RHTAPBUGS-427
in fact you're now able to remove the 3-rd  paty Openshift environment and SEB is still there. Is that expected

**Pull request merged**